bioconda_utils/lint_functions.py

```diff
@@ -236,6 +236,34 @@ def should_not_be_noarch(recipe, meta, df):
         }
 
 
+def setup_py_install_args(recipe, meta, df):
+    if 'setuptools' not in _get_deps(meta, 'build'):
+        return
+
+    err = {
+        'needs_setuptools_args': True,
+        'fix': ('add "--single-version-externally-managed --record=record.txt" '
+                'to setup.py command'),
+    }
+
+    script_line = meta.get('build', {}).get('script', '')
+    if (
+        'setup.py install' in script_line and
+        '--single-version-externally-managed' not in script_line
+    ):
+        return err
+
+    build_sh = os.path.join(recipe, 'build.sh')
+    if not os.path.exists(build_sh):
+        return
+
+    for line in open(build_sh):
```

Comment:
I would read the entire file here at once. We don't know if setup.py and --single-version... is on the same line.